File: cases/taint/typescript/branch-join-negative/BranchJoinNegative.ts
function dfb_source(): string { // DFB-SOURCE: branch-join-input
  return "tainted";
}

function dfb_sink(value: string): void {} // DFB-SINK: branch-join-sink

function run(overwrite: boolean): void {
  let value: string = dfb_source();
  if (overwrite) {
    value = "clean";
  } else {
    value = "clean";
  }
  // DFB-WITNESS: branch-join-value
  dfb_sink(value);
}

File: cases/taint/scala/branch-join-negative/BranchJoinNegative.scala
package dataflowbench

object BranchJoinNegative {
  def dfb_source(): Int = { // DFB-SOURCE: branch-join-input
    1
  }

  def dfb_sink(value: Int): Unit = {} // DFB-SINK: branch-join-sink

  def run(overwrite: Boolean): Unit = {
    var value = dfb_source()
    if (overwrite) {
      value = 0
    } else {
      value = 0
    }
    // DFB-WITNESS: branch-join-value
    dfb_sink(value)
  }
}
